execution/executor/src/chunk_executor.rs

```diff
@@ -295,6 +285,68 @@ impl<V: VMExecutor> ChunkExecutorInner<V> {
     }
 }
 
+#[cfg(not(feature = "consensus-only-perf-test"))]
+fn verify_chunk(
+    txn_list_with_proof: TransactionListWithProof,
+    verified_target_li: &LedgerInfoWithSignatures,
+    first_version_in_request: Option<u64>,
+    latest_view: &ExecutedTrees,
+    num_txns: usize,
+) -> Result<
+    (
+        aptos_types::proof::TransactionInfoListWithProof,
+        usize,
+        Vec<Transaction>,
+    ),
+    anyhow::Error,
+> {
+    // Verify input transaction list
+    txn_list_with_proof.verify(verified_target_li.ledger_info(), first_version_in_request)?;
+
+    let txn_list = txn_list_with_proof.transactions;
+    let txn_info_with_proof = txn_list_with_proof.proof;
+
+    // Skip transactions already in ledger
+    let txns_to_skip = txn_info_with_proof.verify_extends_ledger(
+        latest_view.txn_accumulator().num_leaves(),
+        latest_view.txn_accumulator().root_hash(),
+        first_version_in_request,
+    )?;
+
+    let mut transactions = txn_list;
+    transactions.drain(..txns_to_skip as usize);
+    if txns_to_skip == num_txns {
+        info!(
+            "Skipping all transactions in the given chunk! Num transactions: {:?}",
+            num_txns
+        );
+    }
+
+    Ok((txn_info_with_proof, txns_to_skip, transactions))
+}
+
+#[cfg(feature = "consensus-only-perf-test")]
+fn verify_chunk(
+    txn_list_with_proof: TransactionListWithProof,
+    _verified_target_li: &LedgerInfoWithSignatures,
+    _first_version_in_request: Option<u64>,
+    _latest_view: &ExecutedTrees,
+    _num_txns: usize,
+) -> Result<
+    (
+        aptos_types::proof::TransactionInfoListWithProof,
+        usize,
+        Vec<Transaction>,
+    ),
+    anyhow::Error,
+> {
```

Comment:
nit: maybe add a comment in here explicitly calling out that no validation is being done